WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -155,6 +168,43 @@ class AuthenticationManager: Authentication {
 
         return false
     }
+
+    /// Injects `loggedOutAppSettings`
+    ///
+    func setLoggedOutAppSettings(_ settings: LoggedOutAppSettingsProtocol) {
+        loggedOutAppSettings = settings
+    }
+
+    /// Checks the given site address and see if it's valid
+    /// and returns an error view controller if not.
+    func errorViewController(for siteURL: String, with matcher: ULAccountMatcher, navigationController: UINavigationController) -> UIViewController? {
+
+        /// Account mismatched case
+        guard matcher.match(originalURL: siteURL) else {
+            DDLogWarn("⚠️ Present account mismatch error for site: \(String(describing: siteURL))")
+            let viewModel = WrongAccountErrorViewModel(siteURL: siteURL, showsConnectedStores: matcher.hasConnectedStores)
+            let mismatchAccountUI = ULAccountMismatchViewController(viewModel: viewModel)
+            return mismatchAccountUI
+        }
+
+        /// No Woo found
+        if let matchedSite = matcher.matchedSite(originalURL: siteURL),
+           matchedSite.isWooCommerceActive == false {
+            let viewModel = NoWooErrorViewModel(
+                siteURL: siteURL,
+                showsConnectedStores: matcher.hasConnectedStores,
+                showsInstallButton: matchedSite.isJetpackConnected,
+                onSetupCompletion: { [weak self] siteID in
+                    guard let self = self else { return }
+                    self.startStorePicker(with: siteID, in: navigationController)
```

Comment:
Good catch! We don't need the closure at app start, but it may still be necessary during the login flow, so I added it back in 82d97cb5d660db38bf6002aeee769d8bcd38def0.